### ISSUE TITLE
Create custom map report templates with INGC brand

### ISSUE THREAD

**Issue by gubuntu** [kartoza/geosafe#300]
Do English re-design before #299 and then translate that into PT in #299. 

Use the design outputs from https://github.com/kartoza/geonode/issues/420

child of #306

**Comment by gubuntu** [comment]
Look carefully at existing InaSAFE reports with Indonesia branding. 

Discuss with INGC what changes they would like to make.

- Disclaimer
- logo
- layout
- other aspects?

**Comment by gubuntu** [comment]
@myarjunar how far have you got with this? 

IF none of the English is translated to  PT yet please send me the template(s) so I can get the strings translated.

**Comment by myarjunar** [comment]
I have translated what's in the translated documents. But some strings are no longer relevant with InaSAFE v4. So there must be a lot of strings left to be translated.

**Comment by gubuntu** [comment]
@myarjunar have you done the INGC branding and layout of the localised map template(s)? Please point me to it so I can do the translations.

**Comment by myarjunar** [comment]
@gubuntu no, because I'm working on to make geosafe able to use a custom template first. Should I do the template customization too? There is no mockup for report template with INGC branding though. The text would not change much from the original report except for the disclaimer text (if they decided to add one).

**Comment by gubuntu** [comment]
Can't we just modify the existing template in the first place? All the design materials are available in Drive. I'm sure @raisserafael organised a local disclaimer, I'll try to find it. 

And doesn't InaSAFE already have locale logic e.g. if we set it to PT locale then it will use PT translations, _pt templates etc. ?

The only new thing we need to do which is lower priority is like the web interface in Realtime to upload replacement templates.

**Comment by myarjunar** [comment]
@gubuntu yes we can. I will just apply the INGC theme into the template. But again, we need some kind of mechanism to use custom template (like template uploader in realtime). Because InaSAFE will not automatically use locale related templatem. It only works for InaSAFE translation. cmiiw @lucernae

**Comment by lucernae** [comment]
After reading the discussions I summarize the following:

1. Some strings in InaSAFE still needs to be translated via transifex because it will not show up on the report template files (QPT files), but do show up in the end report because it was generated dynamically.
2. If we exclude feature for web interface upload like in Realtime then we need to put the custom template in `ingc_geonode_theme` repo as a static resources
3. I don't know how to do it in a generic way now, but we could implement a mechanism to take which report template is used for which locale using a settings file in `ingc_geondoe_theme` repo and have `geosafe` repo implement a way to take that information and run the analysis with the given report template
4. Yes, I would suggest for us to just take the map template from default template in InaSAFE and translate it (if there are any raw strings there, but I doubt it). For most design changes you had to modify the QPT files anyway.
5. I'm not sure where the disclaimer is currently resides. CMIIW @myarjunar do we have to load different QgsSettings for that to work? If it's just static maybe we can just put it on the QPT?

**Comment by myarjunar** [comment]
![selection_289](https://user-images.githubusercontent.com/11134669/46845847-6aff2880-ce07-11e8-8969-38c58a031b6b.png)
@gubuntu I only apply INGC skin color and change some logos. @lucernae yes the disclaimer is static, so I will just wait for INGC version of it.

**Comment by gubuntu** [comment]
@myarjunar that template looks great. 

The 'discaimer' that @raisserafael had sent was actually the background paragraph for the home page, which is not really a disclaimer. Please post the English BNPB / generic disclaimer text here and I'll modify it for MZ and translate it to PT

**Comment by lucernae** [comment]
Beautiful @myarjunar !

**Comment by myarjunar** [comment]
@gubuntu here is InaSAFE default disclaimer text
```
InaSAFE has been jointly developed by the Indonesian Government-BNPB, the Australian Government, the World Bank-GFDRR and independent contributors. These agencies and the individual software developers of InaSAFE take no responsibility for the correctness of outputs from InaSAFE or decisions derived as a consequence.
```

**Comment by gubuntu** [comment]
MZ versions

EN

`InaSAFE was jointly developed by the Indonesian Government-BNPB, the Australian Government, the World Bank-GFDRR and independent contributors. These agencies, the individual software developers of InaSAFE and the Government of Mozambique take no responsibility for the correctness of outputs from InaSAFE or decisions derived as a consequence.`

PT

`O InaSAFE foi desenvolvido em conjunto pelo governo da Indonésia - BNPB, o governo Australiano, o Banco Mundial - GFDRR e contribuintes independentes. Estas agências, os desenvolvedores individuais de software da InaSAFE e o Governo de Moçambique não assumem qualquer responsabilidade pela correcção dos resultados da InaSAFE ou decisões derivadas como consequência.`

**Comment by myarjunar** [comment]
We are now able to use a custom template in GeoSAFE. The template should come along with the custom geonode theme repo. Yet I'm still not able to make the basemap work. 
![selection_291](https://user-images.githubusercontent.com/11134669/46934682-ce48bf00-d082-11e8-8d99-c7082bd7119a.png)

**Comment by myarjunar** [comment]
> We are now able to use a custom template in GeoSAFE. The template should come along with the custom geonode theme repo. Yet I'm still not able to make the basemap work.
> ![selection_291](https://user-images.githubusercontent.com/11134669/46934682-ce48bf00-d082-11e8-8d99-c7082bd7119a.png)

@gubuntu could you help me translate the rest of the text in the map report to Portuguese as well? I will change every static text in the template to Portuguese for PT template.

**Comment by gubuntu** [comment]
- [x]  Disclaimer: Aviso legal
- [x] disclaimer text: you have new EN and PT in https://github.com/kartoza/geosafe/issues/300#issuecomment-429270980
- [x] Legend: Legenda
- [ ] Number of structures [where would that get translated?] 
- [x] Analysis Information: Informação de análise
- [x] Time: Hora
- [x] Note: Nota
- [x] Note text: Esta avaliação é um guia - recomendamos enfaticamente que você establece a verdade os resultados mostrados aqui antes de implementar medidas ou pessoal
- [x] Data source: Fonte de dados
- [x] Hazard: Perigo
- [x] Exposure: Exposição
- [x] Aggregation: Agregação
- [ ] Analysis: Análise
- [x] Software: Programa
- [x] Reference: Referência
- [x] Contributors: Contribuinte

**Comment by gubuntu** [comment]
@raisserafael please test and review in production

**Comment by gubuntu** [comment]
@myarjunar all the translations above have not been applied

[gavin flood on minplants_map.pdf](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/53ba46167953d2fd0f723be0/125516c66e82c728ace21e0d46ef958c2a8cc3af83e6b047e452fed4244f29632505ad3f63baee29af0037431f1d11f6140b1b56f6eb8c23f7b46878dc085fec84680a74936f11acbed618eb05bf32975dc7e7e2de139db17770a599855d3f190f198cfb5c35a7994fc85bf44fe35f18.pdf)

**Comment by NyakudyaA** [comment]
> * [x]   Disclaimer: Aviso legal
> * [x]  disclaimer text: you have new EN and PT in [#300 (comment)](https://github.com/kartoza/geosafe/issues/300#issuecomment-429270980)
> * [x]  Legend: Legenda
> * [ ]  Number of structures [where would that get translated?]

This has not yet been done . Are we going to do it ?
![screenshot 2018-11-13 at 12 00 09](https://user-images.githubusercontent.com/2510900/48406447-2ab00480-e73d-11e8-916b-a63372b63ad1.png)

![screenshot 2018-11-13 at 12 00 27](https://user-images.githubusercontent.com/2510900/48406542-75318100-e73d-11e8-8769-170e42b02f26.png)


> * [x]  Analysis Information: Informação de análise
> * [x]  Time: Hora
> * [x]  Note: Nota
> * [x]  Note text: Esta avaliação é um guia - recomendamos enfaticamente que você establece a verdade os resultados mostrados aqui antes de implementar medidas ou pessoal
> * [x]  Data source: Fonte de dados
> * [x]  Hazard: Perigo
> * [x]  Exposure: Exposição
> * [x]  Aggregation: Agregação
> * [x]  Analysis: Análise

The text specified here and the one in the report is different. Which one should it be ?.
![screenshot 2018-11-13 at 12 11 38](https://user-images.githubusercontent.com/2510900/48406495-4d421d80-e73d-11e8-8198-c565f04972cc.png)

> * [x]  Software: Programa
> * [x]  Reference: Referência
> * [x]  Contributors: Contribuinte

**Comment by NyakudyaA** [comment]
@myarjunar Please see my comments above for this issue

**Comment by myarjunar** [comment]
@NyakudyaA those strings will be translated in InaSAFE after new release. 

> The text specified here and the one in the report is different. Which one should it be ?.

I don't understand this part..

**Comment by NyakudyaA** [comment]
> @NyakudyaA those strings will be translated in InaSAFE after new release.

Ok cool
> 
> > The text specified here and the one in the report is different. Which one should it be ?.
> 
> I don't understand this part..

@myarjunar what I mean was that in the report it specifies `Nome de analise` but this ticket just says `analise`

**Comment by Vries14** [comment]
@myarjunar could you please confirm if 'Nome de analise' is the translation that should be used for 'Analysis'?

**Comment by NyakudyaA** [comment]
The first one seems correct .
![screenshot 2018-11-24 at 08 25 31](https://user-images.githubusercontent.com/2510900/48965277-99edea00-efc2-11e8-9768-ab528ecde37b.png)

@myarjunar Why do we do a partial string translation here. `affected by flood event` should also be in PT
![screenshot 2018-11-24 at 08 25 42](https://user-images.githubusercontent.com/2510900/48965278-99edea00-efc2-11e8-9b48-90fb37e92f49.png)
